src/buckets/components/context/csvUploaderProvider.tsx

```diff
@@ -40,19 +41,23 @@ const MAX_CHUNK_SIZE = 1750
 
 export const CsvUploaderProvider: FC<Props> = React.memo(({children}) => {
   const [progress, setProgress] = useState(0)
-  const {bucketID} = useParams()
+  const [bucketID, setBucket] = useState('')
```

Comment:
you're making new sources of truth again, this time around "what is the currently selected bucket". this is bad. i'd either figure out why useParam didn't work, or pass it in as a function parameter for uploadCsv, so that you can fetch it someplace else.

Reply:
I'm pretty sure the reason it didn't work was because we were calling useParams in the provider, which is wrapping the route. We could nest things down?